[tfgan] Respect use_loss_summaries in GANEstimator

lgeiger opened the pull request:
Since the refactor done in 47dea684efa41981e10299c2737317c504ce41af the `use_loss_summaries` argument of GANEstimator isn't respected anymore.
This PR restores the original behavior and passes `use_loss_summaries` down to the loss functions.

tensorflowbutler commented:
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

caisq commented:
Can you review this please, @joel-shor ?

joel-shor commented on tensorflow/contrib/gan/python/estimator/python/gan_estimator_impl.py:
```diff
@@ -214,12 +214,18 @@ def _get_gan_model(
     estimator_spec = model_fn_lib.EstimatorSpec(
         mode=mode, predictions=gan_model.generated_data)
   else:
+    if 'add_summaries' in inspect.getargspec(generator_loss_fn).args:
```
The previous code didn't have this check. Furthermore, the losses test checks that this arg is present: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/gan/python/losses/python/losses_impl_test.py#L158

How would you feel about removing this argument?

lgeiger replied:
I'm happy to remove this check.
I wasn't sure since the estimator tests failed otherwise due to the missing argument in the dummy loss used there.

joel-shor replied:
Ah good point. Would you mind adding that argument to the dummy loss function? Sorry for that, and great catch!

joel-shor reviewed (changes requested):
Please see the comment I made about checking for `add_summaries` using `inspect`.

lgeiger committed:
[tfgan] Respect use_loss_summaries in GANEstimator

Since the refactor done in 47dea684efa41981e10299c2737317c504ce41af the `use_loss_summaries` argument of GANEstimator isn't respected anymore. This PR restores the original behavior and passes `use_loss_summaries` down to the loss functions.

lgeiger commented:
@joel-shor Sorry for the delay, I removed the check for `add_summaries` using `inspect`.

joel-shor commented:
Great catch, and good fix! Thanks, @lgeiger

joel-shor commented:
@lgeiger would you mind investigating the test failures?

lgeiger commented:
This looked like random time-outs, not related to this PR. Thanks for merging 👍
